x/leverage/types/keys.go

```diff
@@ -88,3 +99,37 @@ func CreateCollateralSettingKey(borrowerAddr sdk.AccAddress, uTokenDenom string)
 	key = append(key, []byte(uTokenDenom)...)
 	return append(key, 0) // append 0 for null-termination
 }
+
+// CreateReserveAmountKey returns a KVStore key for getting and setting the amount reserved of a a given token.
+func CreateReserveAmountKey(tokenDenom string) []byte {
+	// reserveamountprefix | denom | 0x00
+	var key []byte
+	key = append(key, KeyPrefixReserveAmount...)
+	key = append(key, []byte(tokenDenom)...)
+	return append(key, 0) // append 0 for null-termination
+}
+
+// CreateLastInterestTimeKey returns a KVStore key for getting and setting the amount reserved of a a given token.
+func CreateLastInterestTimeKey() []byte {
+	// lastinterestprefix
+	var key []byte
+	key = append(key, KeyPrefixLastInterestTime...)
+	return key
+}
+
+// CreateExchangeRateKey returns a KVStore key for getting and setting the token:uToken rate for a a given token.
+func CreateExchangeRateKey(tokenDenom string) []byte {
+	// exchangeRatePrefix | denom | 0x00
+	var key []byte
+	key = append(key, KeyPrefixExchangeRate...)
+	key = append(key, []byte(tokenDenom)...)
+	return append(key, 0) // append 0 for null-termination
+}
+
+// CreateExchangeRateKeyNoDenom returns a safe copy of exchangeRatePrefix
```

Comment:
Why is this needed? Can't you just call and directly use `KeyPrefixExchangeRate`

Reply:
Note: As discussed on slack, returning a copy with a new underlying array for slice safety.